test/jasmine/tests/range_slider_test.js

```diff
@@ -0,0 +1,229 @@
+
+            slide(start, 400, end, 400);
+
+            var rangeDiff3 = gd._fullLayout.xaxis.range[1] - gd._fullLayout.xaxis.range[0];
+
+            expect(rangeDiff3).toBeLessThan(rangeDiff2);
+        });
+    });
+
+
+    describe('when not specified as visible', function() {
+        beforeEach(function(done) {
+            gd = createGraphDiv();
+
+            Plotly.plot(gd, [{ x: [1,2,3], y: [2,3,4] }], {}).then(function() {
+                rangeSlider = document.getElementsByClassName('range-slider')[0];
+                done();
+            });
+        });
+
+        afterEach(destroyGraphDiv);
+
+        it('should not be added to the DOM by default', function() {
+            expect(rangeSlider).not.toBeDefined();
+        });
+    });
+
+    describe('supplyLayoutDefaults function', function() {
+
+        it('should not mutate layoutIn', function() {
+            var layoutIn = { xaxis: { rangeslider: { visible: true }}, yaxis: {}},
+                layoutOut = { xaxis: { rangeslider: {}}, yaxis: {}},
+                expected = { xaxis: { rangeslider: { visible: true }}, yaxis: {}};
+
+            RangeSlider.supplyLayoutDefaults(layoutIn, layoutOut);
+
+            expect(layoutIn).toEqual(expected);
+        });
+
+        it('should ignore if there is no xaxis or rangeslider properties', function() {
+            var layoutIn = {},
+                layoutOut = {};
+
+            RangeSlider.supplyLayoutDefaults(layoutIn, layoutOut);
+
+            expect(layoutOut).toEqual(layoutIn);
+        });
+
+        it('should set defaults if rangeslider.visible is true', function() {
+            var layoutIn = { xaxis: { rangeslider: { visible: true }}, yaxis: {}},
+                layoutOut = { xaxis: { rangeslider: {}}, yaxis: {}},
+                expected = {
+                    xaxis: {
+                        rangeslider: {
+                            visible: true,
+                            height: 0.15,
+                            backgroundcolor: '#ffffff',
+                            borderwidth: 1,
+                            bordercolor: 'transparent'
+                        }
+                    },
+                    yaxis: {
+                        fixedrange: true
+                    }
+                };
+
+            RangeSlider.supplyLayoutDefaults(layoutIn, layoutOut);
+
+            expect(layoutOut).toEqual(expected);
+        });
+
+        it('should set defaults if properties are invalid', function() {
+            var layoutIn = { xaxis: { rangeslider: {
+                    visible: 'invalid',
+                    height: 'invalid',
+                    backgroundcolor: 42,
+                    bordercolor: 42,
+                    borderwidth: 'superfat'
+                }}, yaxis: {}},
+                layoutOut = { xaxis: { rangeslider: {}}, yaxis: {}},
+                expected = { xaxis: { rangeslider: {
+                    visible: false,
+                    height: 0.15,
```

Comment:
ICYMI :arrow_double_up: , the attributes shouldn't be coerced when `visible` isn't `true`.